### PR TITLE
handle json signup questionnaire

### LINKED ISSUE #236: Need to handle json custom fields
## Details
We have a new TC2 meta_agency field `signup_questionnaire` which is json.

Currently our only custom field options are `str`, `int`, `bool`, `fk_field`

#### Idea 1
create a new custom field type `json`
then json stringify on the way to PD
then on the way back from PD to Hermes, turn the json string back into json

#### Idea 2
I am not sure how but separate each json question into its own field

### PR DESCRIPTION
**Pull request by PrenSJ2**
Close #236 

### Technical Description

Take the tc2 `client.meta_agency.signup_questionnaire` json value from tc2, save it to the hermes Company then json dump it into a string for PD

also ignore changes to the `signup_questionnaire` in PD

### Testing

- setup as before - look at read me
- add a new custom field (check updated readme)
- signup as new company, remember to fill in to the signup questionnaire (you will need to clear application cache for signup page and flall)
- check signup questionnaire is in hermes company and pd Org

### CODE REVIEW

**Review by DanJezeph15 — COMMENTED** [review]
* [ ] One scenario that the signup questionnaire doesn't go through is if the form isn't filled out before the company is created. If you wait for the company to be created and then fill out the form and submit, the hermes and pipedrive users don't have the data for the questionnaire until a new webhook is sent (eg. you edit the client). In the real thing tho I doubt people will be that slow filling it out and also I believe it takes longer to setup on live rather than locally.

File: migrations/models/5_20240626160514_update.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from tortoise import BaseDBAsyncClient
+
+
+async def upgrade(db: BaseDBAsyncClient) -> str:
+    return """
+        ALTER TABLE "company" ADD "signup_questionnaire" JSONB;
+        """
```
[the remark this answers is not in the document]
Comment (PrenSJ2) [reply migrations/models/5_20240626160514_update.py:7]: dont believe so, as this is they way the other aerich migrations work

**Comment by PrenSJ2** [comment]
> * [ ]  One scenario that the signup questionnaire doesn't go through is if the form isn't filled out before the company is created. If you wait for the company to be created and then fill out the form and submit, the hermes and pipedrive users don't have the data for the questionnaire until a new webhook is sent (eg. you edit the client). In the real thing tho I doubt people will be that slow filling it out and also I believe it takes longer to setup on live rather than locally.

like you have mentioned, this is unlikely to happen. only solution would be somehome force another client created or updated webhook to be sent to hermes from TC2 when the form is submitted

**Comment by PrenSJ2** [comment]
Once this has been merged and deployed `aerich upgrade` will need to be run to perform the migration

**Pull request merged**